### PR TITLE
Implement auto clock speed's client<->daemon message sending

### LINKED ISSUE #68: Connect client and daemon

### PR DESCRIPTION
**Pull request by Camerooooon**
Closes #68

### CODE REVIEW

**Comment by Camerooooon** [comment]
To test use `socat - UNIX-CONNECT:/tmp/rclc.sock` to hook into the daemon as the "client" after starting the daemon. You should be able to type packets however without the proper syntax parsing will fail.

**Comment by Camerooooon** [comment]
@JakeRoggenbuck this is a big pull request, all feedback please! This will form the base for our client to client communication as well when we get to implementing it!

File: client/src/hook.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            Ok(stream) => stream,
+            Err(e) => {
+                notif(
+                    "Client Error",
+                    &format!("Could not serialize packet: {:?}", e),
+                );
+                return;
+            }
+        };
+        let mut to_send = match serialize_message_client_to_daemon(ClientToDaemonMsg::Test(
```
Comment (JakeRoggenbuck) [inline client/src/hook.rs:20]: Is this the protocol start or just a test for now?

File: client/src/hook.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            "The quick brown fox jumped over the lazy dogs".to_string(),
+        )) {
+            Ok(b) => b,
+            Err(e) => {
+                notif(
```
Comment (JakeRoggenbuck) [inline client/src/hook.rs:25]: Should we log this somewhere else?

File: client/src/hook.rs (hunk 1 above)
Comment (JakeRoggenbuck) [reply client/src/hook.rs:20]: Oh i see, this is the Test()

File: daemon/src/listen.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                notif(
+                    "Daemon Error",
+                    &format!("Failed to bind to {}: {}", path, e),
+                );
+                return;
+            }
+        };
+
+        // Set the permissions on the sock
+        std::fs::set_permissions(path, std::fs::Permissions::from_mode(0o777)).ok();
```
Comment (JakeRoggenbuck) [inline daemon/src/listen.rs:36]: This lets other users read messages on accounts that aren't theirs. Maybe remove the groups perm.

File: daemon/src/listen.rs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            };
+            let message = match parse_message_client_to_daemon(actual_line.into_bytes()) {
+                Ok(p) => p,
+                Err(e) => {
+                    notif("Stream Error", &format!("Received malformed packet: {}", e));
+                    ClientToDaemonMsg::Unknown
+                }
+            };
+
+            println!("Message: {:?}", message);
```
Comment (JakeRoggenbuck) [inline daemon/src/listen.rs:91]: :eyes:

File: daemon/src/listen.rs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+            // TODO: This would be an excellent use for a macro
```
Comment (JakeRoggenbuck) [inline daemon/src/listen.rs:93]: It would be

File: daemon/src/listen.rs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            let to_send = match serialize_message_daemon_to_client(DaemonToClientMsg::Test(
+                "The quick brown fox jumped over the lazy dogs".to_string(),
+            )) {
+                Ok(b) => b,
+                Err(e) => {
+                    notif(
```
Comment (JakeRoggenbuck) [inline daemon/src/listen.rs:99]: Yea, we should definitely have a log file

Comment (JakeRoggenbuck) [reply daemon/src/listen.rs:99]: That can be future changes

**Review by JakeRoggenbuck — DISMISSED** [review]
Works!

File: client/src/hook.rs (hunk 1 above)
Comment (Camerooooon) [reply client/src/hook.rs:20]: Nope just a test, we'll add some hooking logic packets later on.

File: common/src/client_daemon.rs [hunk 7]
```diff
@@ -15,11 +42,47 @@ pub struct Message {
...
+}
+
+/// Converts a DaemonToClientMsg object into bytes to be sent over a stream
+pub fn serialize_message_daemon_to_client(
+    message: DaemonToClientMsg,
+) -> Result<Vec<u8>, postcard::Error> {
+    //let out: heapless::Vec<u8, 128> = postcard::to_vec(&message)?;
+    let out: Vec<u8> = postcard::to_vec::<DaemonToClientMsg, 16384>(&message)?.to_vec();
+    // TODO: The number 16384 is the amount of bytes that will be read, if it exceeds this number it will
+    // error. I'm not yet sure how to make this dyanmically alocate so I just set it to a really
```
Comment (adamhutchings) [inline common/src/client_daemon.rs:73]: allocate

File: daemon/src/listen.rs (hunk 3 above)
Comment (Camerooooon) [reply daemon/src/listen.rs:36]: Good idea!

**Review by JakeRoggenbuck — APPROVED** [review]

**Pull request merged**